wolox-books/src/app/services/cart-count.service.ts

```diff
@@ -0,0 +1,19 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject } from 'rxjs/internal/BehaviorSubject';
+import { Observable } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class CartCountService {
+
+  private countSource = new BehaviorSubject<number>(0);
+  currentCount: Observable<number> = this.countSource.asObservable();
+
+  constructor() { }
+
+  changeCount(count: number){
+    event.stopPropagation();
```

Comment:
why this?

Reply:
Since the "Add button" is inside a container that has a router, if i don't add that it would also redirect when the button is pressed and that behaviour is unwanted.